src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/finallocalvariable/InputFinalLocalVariableCheckSwitchExpressions.java

```diff
@@ -132,5 +132,12 @@ void foo4() throws Exception {
         }
 
     }
+
+    private static final int staticValue = 2;
+    private static final int staticField = switch(staticValue) {
+        case 0 -> -1;
+        case 2-> 2;
+        default -> 3;
+    };
```

Comment:
As @romani stated in his items, please create a new UT and input file.  It would be good to show:

 -  violations: definitions where variable is not declared `final` but is never reassigned
 -  acceptable variable definitions (those that are declared `final` and never reassigned)
 - acceptable variable definitions (where the variable is not declared final AND is reassigned).

...  where a switch expression is on RHS of the assignment.   Try to put the code within methods, class blocks, constructors, etc. to make sure that we can successfully check all.

Reply:
You're welcome @nmancus1. I have created a new UT and a new input file with violations and acceptable cases. I've also put a switch expression on RHS of the assignment,... like you suggested

Reply:
Resolved.